lib/src/rmui_bundle_update_suggestors/dart_script_updater.dart

```diff
@@ -39,74 +54,103 @@ class DartScriptUpdater extends RecursiveAstVisitor<void>
       ...Script(pathSubpattern: existingScriptPath)
           .pattern
           .allMatches(stringValue),
-      ...Script(pathSubpattern: newScriptPath).pattern.allMatches(stringValue)
+      ...?(!removeTag
+          ? Script(pathSubpattern: newScriptPath)
+              .pattern
+              .allMatches(stringValue)
+          : null)
     ];
     final relevantLinkTags = [
       ...Link(pathSubpattern: existingScriptPath)
           .pattern
           .allMatches(stringValue),
-      ...Link(pathSubpattern: newScriptPath).pattern.allMatches(stringValue)
+      ...?(!removeTag
+          ? Link(pathSubpattern: newScriptPath).pattern.allMatches(stringValue)
+          : null)
     ];
 
     // Do not update if neither the existingScriptPath nor newScriptPath are in the file.
     if (relevantScriptTags.isEmpty && relevantLinkTags.isEmpty) return;
 
-    // Add type="module" attribute to script tag.
-    for (final scriptTagMatch in relevantScriptTags) {
-      final scriptTag = scriptTagMatch.group(0);
-      if (scriptTag == null) continue;
-      final typeAttributes = getAttributePattern('type').allMatches(scriptTag);
-      if (typeAttributes.isNotEmpty) {
-        final attribute = typeAttributes.first;
-        final value = attribute.group(1);
-        if (value == 'module') {
-          continue;
+    if (removeTag) {
+      [...relevantScriptTags, ...relevantLinkTags].forEach((tag) async {
+        final tagEnd = node.offset + tag.end;
+        final tagStart = node.offset + tag.start;
+        final possibleCommaEnd = node.literal.next.toString() == ',' ? 1 : 0;
+        final isTagSameAsNode =
+            // Check if the only difference between [tag] and [node] is the quotes around [node].
+            tagStart - node.offset <= 1 && node.end - tagEnd <= 1;
+
+        yieldPatch(
+          '',
+          // If [tag] spans the whole string literal in [node], then also include
+          // the quotes and comma in the removal.
+          isTagSameAsNode ? node.offset : tagStart,
```

Comment:
It looks like the script doesn't handle removing the newlines, leaving a blank line between, as shown here: https://github.com/Workiva/over_react_codemod/blob/4df4e0b7db4e8f37bb9d5951789e0278f6844faf/test/executables/react_18_upgrade_test.dart#L126-L131

I think that newline would get preserved after formatting, so I think it'd be worth trying to get rid of in the codemod so we don't have to worry about cleaning those up manually. One trick could be to start the patch at the end of the previous node:
```suggestion
          isTagSameAsNode
              // Remove from the end of the previous token to take any preceding newline with it,
              // so that we don't leave behind an empty line.
              ? node.beginToken.previous?.end ?? node.offset
              : tagStart,
```

Reply:
oh I was thinking formatting would work - thanks for the fix!!